src/Tasks.php

```diff
@@ -1030,4 +1030,53 @@ public function migrateCleanup($opts = ['migrations' => '']) {
       return FALSE;
     }
   }
+
+
+  /**
+   * Runs a "composer update" and pushes results to the "autoupdate" branch.
+   *
+   * This command should be ran after checking out the default branch of a
+   * project. If "composer update" results in any changed files, a force push
+   * is used to ensure that the "autoupdate" branch is only one commit ahead
+   * of the default branch.
+   *
+   * @param string $profile
+   *   A specific profile to update, ex: "thinkshout/bene".
+   *
+   * @return bool
+   *   Whether or not the update succeeded.
+   */
+  public function ciUpdate($profile = NULL) {
+    $exec = $this->taskComposerUpdate()
+      ->option('with-dependencies');
+
+    if ($profile) {
+      $exec->arg($profile);
+    }
+
+    $result = $exec->run();
+
+    if (!$result->wasSuccessful()) return FALSE;
+
+    $result = $this->taskExec('git status -s')
+      ->interactive(FALSE)
+      ->run();
+
+    if (!$result->wasSuccessful()) return FALSE;
+
+    if (empty($result->getMessage())) {
+      $this->say('No updates to commit.');
+      return TRUE;
+    }
+
+    $result = $this->taskExec('git checkout -b autoupdate && git add . && git commit -m "Ran automatic updates." && git push --force -u origin autoupdate')
```

Comment:
Won't you get an error if the autoupdate branch exists already?

Reply:
At this point in execution there is no local branch "autoupdate", even though there may be a remote one. This lets us branch off of develop/master every time and replace "autoupdate"'s commit history to make sure it's always up to date.

Reply:
Ah, gotcha. I guess I assumed a `git checkout` would pull all branches.